client/App/pages/LandingPage/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,90 @@
+import styles from './Carousel.module.scss';
+import CarouselItem from './CarouselItem/CarouselItem';
+import { useRef, useEffect, useState } from 'react';
+import CarouselIndicators from './CarouselIndicators/CarouselIndicators';
+import CarouselControls from './CarouselControls/CarouselControls';
+
+interface carouselProps {
+  slides: { image: string; text: string | undefined }[];
+  interval?: number;
+  controls?: boolean;
+  indicators?: boolean;
+  autoPlay?: boolean;
+  children?: JSX.Element;
+}
+
+const Carousel = ({
+  slides,
+  interval = 5000,
+  controls = false,
+  indicators = false,
+  autoPlay = false,
+  children,
+}: carouselProps) => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const slideInterval = useRef<number>();
+
+  const prev = () => {
+    startSlideTimer();
+
+    const index = currentSlide > 0 ? currentSlide - 1 : slides.length - 1;
+    setCurrentSlide(index);
+  };
+
+  const next = () => {
+    startSlideTimer();
+
+    const index = currentSlide < slides.length - 1 ? currentSlide + 1 : 0;
+    setCurrentSlide(index);
+  };
+
+  const startSlideTimer = () => {
+    if (autoPlay) {
+      stopSlideTimer();
+      slideInterval.current = window.setInterval(() => {
```

Comment:
Better don't assign this to ref. Create slideIntervalId state instead and use it. Fe: 
```suggestion 
const id = setInterval(() => {
        setCurrentSlide((currentSlide) => (currentSlide < slides.length - 1 ? currentSlide + 1 : 0));
      }, interval);

      setIntervalId(id);

```